rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java

```diff
@@ -2442,15 +2442,21 @@ JobValidationData validateJobSignal(@HeaderParam("sessionid") String sessionId,
      *            current session
      * @param jobId
      *            id of the job
+     * @param endpointName
+     *          name of the endpoint
      * @param externalEndpointUrl
      *          the external endpoint url to add to the job endpoint list
+     * @param endpointIconUri
+     *          uri of the icon associated with the endpoint
      */
     @POST
     @Path("jobs/{jobid}/externalendpointurl")
-    @Consumes(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
     @Produces(MediaType.APPLICATION_JSON)
     void addExternalEndpointUrl(@HeaderParam("sessionid") String sessionId, @PathParam("jobid")
-    final String jobId, String externalEndpointUrl) throws RestException;
+    final String jobId, @FormParam("endpointName") String endpointName,
+            @FormParam("externalEndpointUrl") String externalEndpointUrl,
+            @FormParam("endpointIconUri") String endpointIconUri) throws RestException;
```

Comment:
For the endpoint icon, during our discussions we agreed to do the same as for PSA endpoints (use the workflow, icon).
Do we really need the parameter `endpointIconUri`

Reply:
it can be null, in which case the workflow icon will be used

Reply:
and if we have many endpoints and one single wf? they will have the same icon of the main wf which might also not have an icon

Reply:
@fviale @mboussaa Fair enough